comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,42 +8,89 @@ namespace comp110_worksheet_6
 {
 	public enum Mark { None, O, X };
 
-	public class OxoBoard
-	{
-		// Constructor. Perform any necessary data initialisation here.
-		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+    public class OxoBoard
+    {
+        private Mark[,] tictacktoe; // declare a field, which is accessible from everywhere inside OxoBoard
 
-		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        public OxoBoard()
+        {
+            tictacktoe = new Mark[3, 3]; // initialize that field in the constructor
+        }
 
-		// If the specified square is currently empty, fill it with mark and return true.
-		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // Return the contents of the specified square.
+        /*
+        public Mark GetSquare(int x, int y)
+        {
+            if (tictacktoe[x, y] == Mark.O) // comparing int and enum - ERROR
+            {
+                return Mark.O;
+            }
+            else if (tictacktoe[x, y] == Mark.X) // comparing int and enum - ERROR
+            {
+                return Mark.X;
+            }
+            else
+            {
+                return Mark.None;
+            }
+        }
+        */
 
-		// If there are still empty squares on the board, return false.
-		// If there are no empty squares, return true.
-		public bool IsBoardFull()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // you can rewrite this function simply like this
+        public Mark GetSquare(int x, int y)
+        {
+            return (Mark)tictacktoe[x, y]; // with an explicit cast
+        }
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            tictacktoe[x, y] = mark; // with an explicit cast
+
+            return false;
```

Comment:
Should be checking here whether the square was originally empty, returning `false` only if it was not